src/WebJobs.Script.WebHost/Controllers/HostController.cs

```diff
@@ -197,21 +197,29 @@ public async Task<IActionResult> Drain([FromServices] IScriptHostManager scriptH
                 return StatusCode(StatusCodes.Status503ServiceUnavailable);
             }
 
-            await _drainSemaphore.WaitAsync();
+            try
+            {
+                await _drainSemaphore.WaitAsync(cancellation);
```

Comment:
I think we'd want the same handling for the resume API as well, right? We have the same issue on that API.

Reply:
hmm yeah. I wasn't going to originally, but I think thats a good idea to include it. We could have a resume call start at some random time when we thought it was draining.